### ISSUE TITLE
Create SIEM apps

### ISSUE THREAD

**Issue by frikky** [frikky/Shuffle-apps#21]
Using the App creator, OpenAPI or Python directly, create any of the following apps (or other ones!):

Splunk
QRadar
ArcSight
Elasticsearch (ELK)
Graylog
Logpoint
MDATP
Azure Sentinel
Sumologic
Qualys
RSA NetWitness
Datadog
Security onion

**Comment by pooki3bear** [comment]
Which functions would be included in a minimum product for SIEM (other than on-demand or prepared search)?

**Comment by frikky** [comment]
@pooki3bear I don't want to say that any "minimum product" is required to be added as app necessarily. For SIEM, it initially would just be search. 

What would be interesting though, would be to find out how to use Sigma to create a good integration for either one of these :+1: 

I can share a spreadsheet if you'd like more insight into what we have outlined

**Comment by dhaval055** [comment]
https://github.com/Shuffle/openapi-apps/blob/master/sumologic-api.yaml

**Comment by dhaval055** [comment]
| No. | Tool | Accessibility | Is a demo required? | APIs
|-----|--------|-------------|-----------|----|
|1 | LogPoint | No direct access available. | Yes | [docs](https://docs.logpoint.com/docs/logpoint-api-reference/en/latest/)
|2| RSA NetWitness | No direct access available. | Yes | [docs](https://community.netwitness.com/t5/netwitness-platform-online/restful-api-user-guide-for-rsa-netwitness-platform-11-x/ta-p/565293)
|3| Logrhythm| No direct access available. | Yes | [docs](https://docs.logrhythm.com/docs/lrapi/rest-api)
|4| Securonix | No direct access available. | Yes | [docs](https://documentation.securonix.com/onlinedoc/Content/Cloud/Content/SNYPR%206.3/Web%20Services/_6.3%20Web%20Services_Intro.htm)
|5| Seceon | No direct access available. | Yes | 
|6| ManageEngine EventLog Analyzer | **APIs not available at the moment.** | No |
|7| ExaBeam | No direct access available. | Yes |[reference](https://community.exabeam.com/s/article/Exabeam-APIs-Overview-and-Context-Table-Demo)
|8| Fluency | No free access available. | No |[docs](https://api.fluencysecurity.com/), [Postman collecton](https://www.postman.com/fluency/workspace/fluency-s-public-workspace/collection/1117493-67acddf8-41bb-4362-b7fb-ec2500db3dd8?ctx=documentation) |

**Comment by frikky** [comment]
> for example if i have my own siem how do i push logs to shuffle so that i can build my SOAR

Hey,

there's quite a few ways, but the main things are:
- Can you do alert forwarding, e.g. with webhooks?
- Do you have a search API?

**Comment by winhigh** [comment]
No. I'm new to this tool could you please let us know what are the possible ways to push my logs to shuffle interface.

search API for , To be honest i need to learn everything 

do you have any possible ways to redirect my another system logs to shuffle , if successfully redirect also , how do i see those logs in shuffle so that i can co relate with other tools like yara.

**Comment by frikky** [comment]
> No. I'm new to this tool could you please let us know what are the possible ways to push my logs to shuffle interface.
> 
> search API for , To be honest i need to learn everything
> 
> do you have any possible ways to redirect my another system logs to shuffle , if successfully redirect also , how do i see those logs in shuffle so that i can co relate with other tools like yara.

We don't typically deal with logs directly, and instead focus on alerts from the SIEM. In this case though, I'd do something like this if I were to handle logs directly with Shuffle tho (we are planning for this ;))

1. Set up a syslog listener (e.g. with Tenzir)
2. When syslogs are found, bucket them
3. Forward to Shuffle over HTTP with a Webhook when you got e.g. 1000 logs bucketed

Shuffle itself isn't meant for this kind of thing, so we suggest you use a SIEM and forward alerts instead :)

**Comment by winhigh** [comment]
hey frikky, 

yeah even i know shuffle isn't designed for logs but i wanted to co relate logs with yara rules or other tool so that it can detect malicious IPs and sing shuffle alerts and automation i can block them.

So basically my idea is to automate my security.

**Comment by winhigh** [comment]
I'm planning to send logs to Shuffle machine using rsyslog or ossec and collect them using webhooks ? 
 
 is it possible ?

**Comment by frikky** [comment]
> I'm planning to send logs to Shuffle machine using rsyslog or ossec and collect them using webhooks ?
> 
> is it possible ?

We got something cooking for this. It's not directly possible right now, but soon~ :)

**Comment by winhigh** [comment]
Hi Frikky,


Actually, I tried sending alerts to shuffle from wazuh tool as you demonstrated in the video but I can't able to get those level three alerts in json.

PS: could you provide me the video, Showcasing alerts after setting with webhooks

***@***.***_abdelhadi_86557/wazuh-shuffle-integration-3dc0b7db439
Followed these instructions.

𝚃𝚑𝚊𝚗𝚔𝚜 𝚊𝚗𝚍 𝚁𝚎𝚐𝚊𝚛𝚍𝚜 ,
𝚅𝚒𝚗𝚊𝚢 𝙺𝚞𝚖𝚊𝚛 𝙷 𝙶
𝙿𝚑𝚘𝚗𝚎: +91 8088688188

[cid:0c7774f1-44ed-4876-ad6c-643e9a7e2911]<http://www.vigyanlabs.com>

Please consider the environment before printing this mail

“Confidential: The information contained in this electronic mail is legally privileged and confidential, intended only for the use of the individual or entity named above. If you have received this email in error, please notify the sender immediately by e-mail and delete this e-mail. Vigyanlabs Innovation Private Limited and its customers do not grant any express or implied rights to you under any patents, copyrights, trademarks, or trade secret information. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication or taking any action in reliance on the contents of this information is strictly prohibited".
________________________________

**Comment by winhigh** [comment]
Hi Frikky, 

Actually, I tried sending alerts to shuffle from wazuh tool as you demonstrated in the video but I can't able to get those level three alerts in json.

PS: could you provide me the video, Showcasing alerts after setting with webhooks

https://medium.com/@ilyes_abdelhadi_86557/wazuh-shuffle-integration-3dc0b7db439
Followed these instructions.

**Comment by winhigh** [comment]
***@***.*** would like to recall the message, "Re: [Shuffle/python-apps] Create SIEM apps (#21)".

**Comment by ayush0033** [comment]
> Using the App creator, OpenAPI or Python directly:

> * [x]  Splunk        - Input Workflow - Search Workflow - Documentation - [Public app](https://shuffler.io/apps/441a2d85f6c1e8408dd1ee1e804cd241?queryID=e97a6a1313f0dc7fe72ffaeb071503d7)
> * [x]  QRadar
> * [ ]  ArcSight
> * [x]  Elasticsearch (ELK)
> * [x]  Logpoint
> * [x]  MDATP
> * [x]  Azure Sentinel
> * [x]  Sumologic
> * [x]  Logz.io
> * [x]  RSA NetWitness - [public app](https://shuffler.io/apps/33cca46bb02f987b120a910e1c09cf37)
> * [x]  [Datadog #301](https://github.com/Shuffle/python-apps/issues/301)
> * [ ]  Logarithm
> * [x]  Security onion
> * [x]  Rapid7 IDR
> * [x]  FortiSIEM
> * [x]  Securonix
> * [x]  [Wazuh #298](https://github.com/Shuffle/python-apps/issues/298)
> * [x]  Seceon
> * [x]  Microsoft Sentinel
> * [ ]  Fluency
> * [x]  CyberShark
> * [x]  ExaBeam
> * [x]  AlertLogic
> * [x]  ManageEngine EventLog Analyzer
> * [x]  New Relic
> * [ ]  Logit.io
> * [x]  Solarwinds Security Event Manager
> * [ ]  Sematext
> * [x]  Servicepilot
